### PR TITLE
doc: matter: improvements to docs

### PR DESCRIPTION
**Pull request by greg-fer**
Added minor improvements across Matter docs.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Review by LuDuda — APPROVED** [review]
Thank you! :+1:

File: samples/matter/lock/README.rst [hunk 1]
```diff
@@ -8,8 +8,8 @@ Matter: Door lock
    :local:
    :depth: 2
 
-This door lock sample demonstrates the usage of the `Matter`_ (formerly Project Connected Home over IP, Project CHIP) application layer to build a door lock device with one basic bolt.
-This device works as a Matter accessory, meaning it can be paired and controlled remotely over a Matter network built on top of a low-power, 802.15.4 Thread network.
+This door lock sample demonstrates the usage of the :ref:`Matter <ug_matter>` (formerly Project Connected Home over IP, Project CHIP) application layer to build a door lock device with one basic bolt.
+This device works as a Matter accessory device, meaning it can be paired and controlled remotely over a Matter network built on top of a low-power, 802.15.4 Thread network.
```
Comment (Damian-Nordic) [inline samples/matter/lock/README.rst:12]: " device" was added only in the lock sample and not in the light bulb.

File: samples/matter/lock/README.rst [hunk 2]
```diff
@@ -111,7 +109,7 @@ SEGGER J-Link USB port:
     Used for getting logs from the device or for communicating with it through the command-line interface.
 
 NFC port with antenna attached:
-    Optionally used for obtaining the commissioning information from the Matter device to start the :ref:`commissioning procedure <matter_lock_sample_remote_control>`.
+    Optionally used for obtaining the commissioning information from the Matter accessory device to start the :ref:`commissioning procedure <matter_lock_sample_remote_control>`.
```
Comment (Damian-Nordic) [inline samples/matter/lock/README.rst:112]: This change (and a few below) hasn't been made for the light bulb sample.

Comment (greg-fer) [reply samples/matter/lock/README.rst:112]: Not sure which edits below you are referring to. Most of them are included in the light bulb sample with the `.. include::` command.

**Review by Damian-Nordic — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**